restapi/src/main/java/io/stargate/web/docsapi/service/JsonSchemaHandler.java

```diff
@@ -0,0 +1,163 @@
+    }
+    this.mapper = mapper;
+  }
+
+  private void clearCacheOnSchemaChange(DocumentDB db) {
+    if (!db.schema().equals(lastCheckedSchema)) {
+      schemasPerCollection.clear();
+      this.lastCheckedSchema = db.schema();
+    }
+  }
+
+  private JsonSchemaResponse reportToResponse(JsonNode schema, ProcessingReport report) {
+    JsonSchemaResponse resp = new JsonSchemaResponse(schema);
+    report.forEach(
+        msg -> {
+          resp.addMessage(msg.getLogLevel(), msg.getMessage());
+        });
+    return resp;
+  }
+
+  public JsonSchemaResponse attachSchemaToCollection(
+      DocumentDB db, String namespace, String collection, JsonNode schema) {
+    ProcessingReport report = schemaFactory.getSyntaxValidator().validateSchema(schema);
+    JsonSchemaResponse resp = reportToResponse(schema, report);
+    if (report.isSuccess()) {
+      writeSchemaToCollection(db, namespace, collection, schema.toString());
+      ImmutableKeyspaceAndTable info =
+          ImmutableKeyspaceAndTable.builder().keyspace(namespace).table(collection).build();
+      schemasPerCollection.put(info, schema);
+      return resp;
+    } else {
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_JSON_SCHEMA_INVALID);
+    }
+  }
+
+  private CompletableFuture<String> getRawJsonSchemaForCollection(
+      DocumentDB db, String namespace, String collection) {
+    BuiltQuery q =
+        db.builder()
+            .select()
+            .column("comment")
+            .from("system_schema", "tables")
+            .where("keyspace_name", Predicate.EQ, namespace)
+            .where("table_name", Predicate.EQ, collection)
+            .build();
+    CompletableFuture<ResultSet> result = q.execute();
+    return result.thenApply(
+        rs -> {
+          List<Row> rows = rs.currentPageRows();
+          if (rows.size() > 0) {
+            String comment = rows.get(0).getString("comment");
+            return comment.isEmpty() ? null : comment;
+          } else {
+            return null;
+          }
+        });
+  }
+
+  public CompletableFuture<JsonSchemaResponse> getJsonSchemaForCollection(
+      DocumentDB db, String namespace, String collection) {
+    return getRawJsonSchemaForCollection(db, namespace, collection)
+        .thenApply(
+            schemaData -> {
+              try {
+                return new JsonSchemaResponse(mapper.readTree(schemaData));
+              } catch (JsonProcessingException e) {
+                return null;
+              }
+            });
+  }
+
+  /**
+   * Attaches a comment to the given C* table that represents a JSON schema.
+   *
+   * @param namespace
+   * @param collection
+   * @param schemaData
+   */
+  private void writeSchemaToCollection(
+      DocumentDB db, String namespace, String collection, String schemaData) {
+    db.builder().alter().table(namespace, collection).withComment(schemaData).build().execute();
```

Comment:
Would it be worth wrapping `schemaData` in another JSON object here, just to make upgrades easier in case we need to store something else in the comment field in the future?

Reply:
Also, should be perhaps wait for schema agreement here to ensure subsequent "get" operations work with the latest data? WDYT?

In this case we can probably remove the "wait for schema agreement" call from the integration test.